sr/robot/astoria.py

```diff
@@ -0,0 +1,294 @@
+
+    @classmethod
+    def load(cls, config_str: Union[str, None] = None) -> "AstoriaConfig":
+        """Load the config."""
+        config_path = cls._get_config_path(config_str)
+        with config_path.open("rb") as fh:
+            return cls.parse_obj(tomllib.load(fh))
+
+
+# Metadata
+
+class RobotMode(Enum):
+    """Running Status of the manager daemon."""
+
+    COMP = "COMP"
+    DEV = "DEV"
+
+
+class Metadata(BaseModel):
+    """
+    Astoria Metadata.
+
+    As the metadata is passed into a templating engine for initial log lines,
+    please do not add nested fields to this schema.
+    """
+    class Config:
+        # Revalidate the model when fields are changed
+        validate_assignment = True
+
+    # From Meta USB
+    arena: str = "A"
+    zone: int = 0
+    mode: RobotMode = RobotMode.DEV
+    marker_offset: int = 0
+    game_timeout: Union[int, None] = None
+    wifi_enabled: bool = True
+
+    # From robot settings file
+    wifi_ssid: Union[str, None] = None
+    wifi_psk: Union[str, None] = None
+    wifi_region: Union[str, None] = None
+
+
+# Process manager fields
+
+class CodeStatus(str, Enum):
+    """Status of the running code."""
+
+    STARTING = "code_starting"
+    RUNNING = "code_running"
+    KILLED = "code_killed"
+    FINISHED = "code_finished"
+    CRASHED = "code_crashed"
+
+
+DiskUUID = NewType("DiskUUID", str)
+
+
+class DiskType(Enum):
+    """Type of disk."""
+
+    USERCODE = "USERCODE"
+    METADATA = "METADATA"
+    NOACTION = "NOACTION"
+
+
+class DiskInfo(BaseModel):
+    """Information about a mounted disk."""
+
+    uuid: DiskUUID
+    mount_path: Path
+    disk_type: DiskType
+
+
+# Manager messages
+
+class ManagerMessage(BaseModel):
+    """Common data that all manager messages output."""
+
+    class Status(Enum):
+        """Running Status of the manager daemon."""
+
+        STOPPED = "STOPPED"
+        RUNNING = "RUNNING"
+
+    status: Status
+    astoria_version: str = astoria_version
+
+
+class ProcessManagerMessage(ManagerMessage):
+    """
+    Status message for Process Manager.
+
+    Published to astoria/astprocd
+    """
+
+    code_status: Union[CodeStatus, None]
+    disk_info: Union[DiskInfo, None]
+    pid: Union[int, None]
+
+
+class MetadataManagerMessage(ManagerMessage):
+    """
+    Status message for Metadata Manager.
+
+    Published to /astoria/astmetad
+    """
+
+    metadata: Metadata
+
+
+# Remote start event
+
+class StartButtonBroadcastEvent(BaseModel):
+    """
+    Schema for a remote start event.
+
+    Trigger the robot code if it is waiting for the start button to be
+    pressed. Does not affect or interact with the physical button as that
+    is handled by the usercode driver.
+    """
+
+    name: ClassVar[str] = "start_button"
+
+    event_name: str
+    sender_name: str
+    priority: int = 0
+
+    def __gt__(self, other: "StartButtonBroadcastEvent") -> bool:
+        return self.priority > other.priority
```

Comment:
Usually better to complete the set if we're implementing comparisons, `@functools.total_ordering` on the class perhaps?

Reply:
28f4c9008c93f1664160a7fb1843a9786f41b83b